File: 차봄(믿고 맡겨조)/backend/src/main/java/kr/co/himedia/security/CustomUserDetails.java
package kr.co.himedia.security;

import kr.co.himedia.entity.User;
import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Collections;
import java.util.UUID;

@Getter
public class CustomUserDetails implements UserDetails {

    private final User user;

    public CustomUserDetails(User user) {
        this.user = user;
    }

    public UUID getUserId() {
        return user.getUserId();
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        // 현재 역할(Role) 필드가 엔티티에 없으므로 기본 ROLE_USER 부여
        return Collections.singletonList(new SimpleGrantedAuthority("ROLE_USER"));
    }

    @Override
    public String getPassword() {
        return user.getPasswordHash();
    }

    @Override
    public String getUsername() {
        return user.getEmail();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: Bask-EAT((주)식회사)/Bask-EAT-Service-Backend/bin/main/com/bask_eat/business_backend/security/CustomUserDetails.kt
package com.bask_eat.business_backend.security

import com.bask_eat.business_backend.model.entity.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class CustomUserDetails(
  private val user: User
) : UserDetails {

  override fun getAuthorities(): Collection<GrantedAuthority> {
    return listOf(SimpleGrantedAuthority("ROLE_${user.role.name}"))
  }

  override fun getPassword(): String = ""

  override fun getUsername(): String = user.id

  override fun isAccountNonExpired(): Boolean = user.isActive

  override fun isAccountNonLocked(): Boolean = user.isActive

  override fun isCredentialsNonExpired(): Boolean = true

  override fun isEnabled(): Boolean = user.isActive

  fun getUser(): User = user
}
